Code/GraphMol/SmilesParse/SmilesWrite.cpp

```diff
@@ -466,35 +453,31 @@ std::string MolToSmiles(const ROMol &mol, bool doIsomericSmiles, bool doKekule,
       std::cout << "----------------------------" << std::endl;
 #endif
 
-    // adding randomness without setting the rootedAtAtom
-    if (doRandom) {
-      if (rootedAtAtom == -1) {
-        // need to find an atom id between 0 and mol.getNumAtoms() exclusively
-        rootedAtAtom = std::rand() % mol.getNumAtoms();
-      }
+    if (doRandom && rootedAtAtom == -1) {
+      // need to find a random atom id between 0 and mol.getNumAtoms()
+      // exclusively
+      rootedAtAtom = std::rand() % mol.getNumAtoms();
     }
 
     std::string res;
     unsigned int nAtoms = tmol->getNumAtoms();
-    UINT_VECT ranks(nAtoms);
+    std::vector<unsigned int> ranks(nAtoms);
     std::vector<unsigned int> atomOrdering;
 
     if (canonical) {
       if (tmol->hasProp("_canonicalRankingNumbers")) {
-        for (unsigned int i = 0; i < tmol->getNumAtoms(); ++i) {
+        for (const auto atom : tmol->atoms()) {
```

Comment:
Would it be worth moving to `detail::common_properties` in types.h ?

Also, I can't find where this property is set. Is it some legacy property from older code?

Reply:
that's an undocumented way to allow you to provide the atom ranking to be used in canonicalization